### PR TITLE
Implement tall-skinny SVD

### PR DESCRIPTION
**Pull request by mrfh92**
## Due Diligence

- General:
    - [x]  **title** of the PR is suitable to appear in the [Release Notes](https://github.com/helmholtz-analytics/heat/releases/latest)
- Implementation:
    - [x] unit tests: all split configurations tested
    - [x] unit tests: multiple dtypes tested
    - [x] documentation updated where needed

## Description

Issue/s resolved: #1456 

## Changes proposed:

-
-
-
-

#### Does this change modify the behaviour of other functions? If so, which?
yes / no

### CODE REVIEW

**Review by ClaudiaComito — CHANGES_REQUESTED** [review]
Looks good to me @mrfh92 , just a few minor changes (see below).  

Does it make sense to add a benchmarking function for SVD already? 

Also, did you want to add a PR description?

Thanks again!

File: heat/core/linalg/svd.py [hunk 1]
```diff
@@ -4,14 +4,200 @@
...
+        else:
+            # this is the distributed, tall skinny case
+            # compute SVD via tall skinny QR
+            if compute_uv:
+                # compute full SVD: first full QR, then SVD of R
+                Q, R = qr(A, mode="reduced", procs_to_merge=qr_procs_to_merge)
+                Utilde_loc, S_loc, Vt_loc = torch.linalg.svd(R.larray, full_matrices=False)
+                Utilde = DNDarray(
+                    Utilde_loc,
+                    Utilde_loc.shape,
```
[the remark this answers is not in the document]
Comment (mrfh92) [reply heat/core/linalg/svd.py]: taken into account

**Review by ClaudiaComito — APPROVED** [review]
Great, thanks a lot @mrfh92 !

**Comment by mrfh92** [comment]
@ClaudiaComito I have re-requested review to let the CI matrix run again after merging main.

**Review by ClaudiaComito — APPROVED** [review]

**Pull request merged**